internal/indexservice/meta_table.go

```diff
@@ -46,31 +63,74 @@ func NewMetaTable(kv kv.TxnKV) (*metaTable, error) {
 }
 
 func (mt *metaTable) reloadFromKV() error {
-	mt.indexBuildID2Meta = make(map[UniqueID]indexpb.IndexMeta)
+	mt.indexBuildID2Meta = make(map[UniqueID]Meta)
+	key := "indexes"
+	log.Debug("LoadWithPrefix ", zap.String("prefix", key))
 
-	_, values, err := mt.client.LoadWithPrefix("indexes")
+	resp, err := mt.client.LoadWithPrefix2(key)
 	if err != nil {
 		return err
 	}
+	var indexBuildIDs []UniqueID
 
-	for _, value := range values {
+	for _, kv := range resp.Kvs {
```

Comment:
Move this loop outside etcd

Reply:
This loop should be kept, and the logic for checking if indexnode is alive should move.